node/src/actors/chain_manager/mining.rs

```diff
@@ -259,6 +259,11 @@ impl ChainManager {
             num_active_identities,
         );
 
+        // `current_retrieval_count` keeps track of how many sources are being retrieved in this
+        // epoch by using a reference-counted atomic counter that can be read and updated safely.
+        let current_retrieval_count = std::sync::Arc::new(std::sync::atomic::AtomicU16::new(0u16));
```

Comment:
Could you import `Arc` and `AtomicU16`?

```suggestion
        let current_retrieval_count = Arc::new(AtomicU16::new(0u16));
```